drivers/ethernet/dwc_xgmac/eth_dwc_xgmac.c

```diff
@@ -0,0 +1,1973 @@
+ * pending interrupts, enables RX and TX, enables interrupts.
+ *
+ * @param dev Pointer to the ethernet device
+ * @retval    0 upon successful completion
+ */
+static int eth_dwc_xgmac_start_device(const struct device *dev)
+{
+	const struct eth_dwc_xgmac_config *dev_conf = (struct eth_dwc_xgmac_config *)dev->config;
+	struct eth_dwc_xgmac_dev_data *dev_data = (struct eth_dwc_xgmac_dev_data *)dev->data;
+	mem_addr_t ioaddr;
+	mem_addr_t reg_addr;
+	uint32_t reg_val;
+
+	if (dev_data->dev_started) {
+		LOG_INF("Eth device already started");
+		return 0;
+	}
+
+	ioaddr = get_reg_base_addr(dev);
+
+	for (uint32_t dma_chnl = 0u; dma_chnl < dev_conf->num_dma_chnl; dma_chnl++) {
+		/* Start the transmit DMA channel */
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TX_CONTROL_OFST);
+		reg_val = sys_read32(reg_addr) | DMA_CHx_TX_CONTROL_ST_SET_MSK;
+		sys_write32(reg_val, reg_addr);
+		/* Start the receive DMA channel */
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RX_CONTROL_OFST);
+		reg_val = sys_read32(reg_addr) | DMA_CHx_RX_CONTROL_SR_SET_MSK;
+		sys_write32(reg_val, reg_addr);
+		/* Enable the dma channel interrupts */
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_INTERRUPT_ENABLE_OFST);
+		reg_val = DMA_CHx_INTERRUPT_ENABLE_NIE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_AIE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_CDEE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_FBEE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_DDEE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_RSE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_RBUE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_RIE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_TBUE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_TXSE_SET(1u) |
+			  DMA_CHx_INTERRUPT_ENABLE_TIE_SET(1u);
+		sys_write32(reg_val, reg_addr);
+		LOG_DBG("%s: Interrupts enabled for DMA Channel %d", dev->name, dma_chnl);
+	}
+	/* Enable the MAC transmit functionality*/
+	reg_val = sys_read32(ioaddr + CORE_MAC_TX_CONFIGURATION_OFST);
+	reg_val |= CORE_MAC_TX_CONFIGURATION_TE_SET(1u);
+	sys_write32(reg_val, (ioaddr + CORE_MAC_TX_CONFIGURATION_OFST));
+	/* Enable the MAC receive functionality*/
+	reg_val = sys_read32(ioaddr + CORE_MAC_RX_CONFIGURATION_OFST);
+	reg_val |= CORE_MAC_RX_CONFIGURATION_RE_SET(1u);
+	sys_write32(reg_val, (ioaddr + CORE_MAC_RX_CONFIGURATION_OFST));
+	/* Enable the MAC Link Status Change Interrupt */
+	reg_val = sys_read32(ioaddr + CORE_MAC_INTERRUPT_ENABLE_OFST);
+	reg_val = CORE_MAC_INTERRUPT_ENABLE_LSIE_SET(1u);
+	sys_write32(reg_val, (ioaddr + CORE_MAC_INTERRUPT_ENABLE_OFST));
+
+#ifdef CONFIG_ETH_DWC_XGMAC_POLLING_MODE
+	k_timer_start(&dev_data->isr_polling_timer,
+		      K_USEC(CONFIG_ETH_DWC_XGMAC_INTERRUPT_POLLING_INTERVAL_US),
+		      K_USEC(CONFIG_ETH_DWC_XGMAC_INTERRUPT_POLLING_INTERVAL_US));
+#else
+	/* If polling mode is configred then start the ISR polling timer */
+	(dev_conf->irq_enable_fn)(dev, true);
```

Comment:
`()` around func pointer is not necessary

Reply:
removed the () around function pointer